### PR TITLE
Matlab assuming responsibility for generating test data

### PR DESCRIPTION
**Pull request by dlg0**
This is the first of what will hopefully be a series of PRs that incrementally replace matlab scripts with data generated in the matlab proper. The data files now also contain the matlab git hash which was used to create them.

### CODE REVIEW

**Comment by grahamlopez** [comment]
github truncates the diff because it is so big. Can you summarize the change? Is it ONLY auto-generated `testing/generated-inputs/**/*.dat` files?

**Comment by dlg0** [comment]
It BOTH updates the testing data files AND removes the matlab scripts from the C++ repo which created them originally AND the part of the `script/gernerate_test_data.m` script which dealt with the element table.

**Comment by grahamlopez** [comment]
How do I know this PR isn't just to pad your github contributor stats?

**Comment by grahamlopez** [comment]
Ah, so this is dealing with the rediculousness that is the element table testing... Did you think of any way to clean that up (such that we don't need 4k+ files)?

**Comment by dlg0** [comment]
Likely yes, but I have not looked at it yet. Are you suggesting that we roll it into this PR?

**Comment by dlg0** [comment]
I was 

> How do I know this PR isn't just to pad your github contributor stats?

I was hoping that contributor status is proportional to the number of changes in each commit :) I figure if I redo all the test files, I'll catch up to Tyler!

**Comment by grahamlopez** [comment]
If it is reasonable, yes. The reason being: this PR touches 4k+ files. If we are going to do that, it would be good to do that as little as possible. If we change them here, and then delete them soon, we should just wait and delete them (and save this modify delta).

If it is too large of a speed bump, then we can wait.

If you decide to do it as part of this PR, put a WIP on it. If you want to wait, create an issue and describe the strategy.

File: src/element_table_tests.cpp [hunk 1]
```diff
@@ -14,49 +14,57 @@ TEST_CASE("element table constructor/accessors/size", "[element_table]")
   options o        = make_options({"-l", std::to_string(levels)});
   element_table t(o, dims);
 
-  std::string test_base = out_base + "_1_1_SG_";
-  for (auto i = 0; i < t.size(); ++i)
   {
```
Comment (grahamlopez) [inline src/element_table_tests.cpp]: Let's just make these scopes named Catch2 sections:
`SECTION("useful description")`

**Comment by dlg0** [comment]
@grahamlopez @bmcdanie 
The basis_test is failing here on a `relaxed_comparision` and I'm unsure why. Perhaps because it is after midnight and my brain isn't operating at full capacity, but after I updated the testing files this test fails, but from what I can see, the old files and the new files have the same number down to the last digit. 

The test fails as ...

```
-------------------------------------------------------------------------------
operator_two_scale function working appropriately - double
  operator_two_scale(4, 3)
-------------------------------------------------------------------------------
/Users/dg6/code/asgard/src/basis_tests.cpp:144
...............................................................................

/Users/dg6/code/asgard/src/basis_tests.cpp:107: FAILED:
  REQUIRE( Approx(*first_it++) .epsilon(std::numeric_limits<TestType>::epsilon() * 1e4) == second_elem )
with expansion:
  Approx( 0.0061763236 ) == 0.0061763236

-------------------------------------------------------------------------------
operator_two_scale function working appropriately - double
  operator_two_scale(5, 5)
-------------------------------------------------------------------------------
/Users/dg6/code/asgard/src/basis_tests.cpp:159
...............................................................................

/Users/dg6/code/asgard/src/basis_tests.cpp:107: FAILED:
  REQUIRE( Approx(*first_it++) .epsilon(std::numeric_limits<TestType>::epsilon() * 1e4) == second_elem )
with expansion:
  Approx( -0.0061763236 ) == -0.0061763236

===============================================================================
test cases:     3 |     2 passed | 1 failed
assertions: 16873 | 16871 passed | 2 failed
```
but i'm not sure if catch2 is printing all the digits which are being compared in `relaxed_comparison`, i.e., in the data files there are more digits than what is being printed here.

**Comment by dlg0** [comment]
@bmcdanie 

Do you know who wrote this comment ?

https://github.com/project-asgard/asgard/blame/develop/testing/scripts/transformations/OperatorTwoScale.m#L32

**Comment by grahamlopez** [comment]
> @bmcdanie
> 
> Do you know who wrote this comment ?
> 
> https://github.com/project-asgard/asgard/blame/develop/testing/scripts/transformations/OperatorTwoScale.m#L32

```
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  30)
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  31) % Unroll the matlab for easier porting
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  32) % WARNING the "porting" version is now unused,
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  33) % there were logical errors that would cause           
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  34) % float indexing                                            
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  35) porting = 0;
6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  36)    
```

**Comment by dlg0** [comment]
> > @bmcdanie
> > Do you know who wrote this comment ?
> > https://github.com/project-asgard/asgard/blame/develop/testing/scripts/transformations/OperatorTwoScale.m#L32
> 
> ```
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  30)
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  31) % Unroll the matlab for easier porting
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  32) % WARNING the "porting" version is now unused,
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  33) % there were logical errors that would cause           
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  34) % float indexing                                            
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  35) porting = 0;
> 6b942278e (Benjamin T. McDaniel 2019-01-27 17:06:49 -0500  36)    
> ```

I think he just added the file from the matlab. But I cannot this comment in the matlab, so am asking if he added the comment during the moving the file from the matlab repo to the C++ repo.

**Comment by grahamlopez** [comment]
reminder: we need to pull the basis tests out from transformations

**Comment by dlg0** [comment]
This is ready to merge.

**Comment by dlg0** [comment]
OK, rebased onto develop. Dealt with the deleted file conflict mid rebase. Force pushed. Seems OK now.

**Comment by dlg0** [comment]
ping ping :) @bmcdanie @grahamlopez
